flow/model/qvalue/avro_converter.go

```diff
@@ -23,104 +35,64 @@ type QValueKindAvroSchema struct {
 //
 // For example, QValueKindInt64 would return an AvroLogicalSchema of "long". Unsupported QValueKinds
 // will return an error.
-//
-// The function currently does not support the following QValueKinds:
-// - QValueKindBit
-//
-// Please note that for QValueKindNumeric and QValueKindETime, RespectNull is always
-// set to false, regardless of the nullable value passed in.
-func GetAvroSchemaFromQValueKind(kind QValueKind, nullable bool) (*QValueKindAvroSchema, error) {
+func GetAvroSchemaFromQValueKind(kind QValueKind) (interface{}, error) {
 	switch kind {
 	case QValueKindString, QValueKindUUID:
-		return &QValueKindAvroSchema{
-			AvroLogicalSchema: "string",
-		}, nil
+		return "string", nil
 	case QValueKindGeometry, QValueKindGeography, QValueKindPoint:
-		return &QValueKindAvroSchema{
-			AvroLogicalSchema: "string",
-		}, nil
+		return "string", nil
 	case QValueKindInt16, QValueKindInt32, QValueKindInt64:
-		return &QValueKindAvroSchema{
-			AvroLogicalSchema: "long",
-		}, nil
+		return "long", nil
 	case QValueKindFloat32:
-		return &QValueKindAvroSchema{
-			AvroLogicalSchema: "float",
-		}, nil
+		return "float", nil
 	case QValueKindFloat64:
-		return &QValueKindAvroSchema{
-			AvroLogicalSchema: "double",
-		}, nil
+		return "double", nil
 	case QValueKindBoolean:
-		return &QValueKindAvroSchema{
-			AvroLogicalSchema: "boolean",
-		}, nil
+		return "boolean", nil
 	case QValueKindBytes, QValueKindBit:
-		return &QValueKindAvroSchema{
-			AvroLogicalSchema: "bytes",
-		}, nil
+		return "bytes", nil
 	case QValueKindNumeric:
-		return &QValueKindAvroSchema{
-			AvroLogicalSchema: map[string]interface{}{
-				"type":        "bytes",
-				"logicalType": "decimal",
-				"precision":   38,
-				"scale":       9,
-			},
+		return AvroSchemaNumeric{
+			Type:        "bytes",
+			LogicalType: "decimal",
+			Precision:   38,
+			Scale:       9,
 		}, nil
 	case QValueKindTime, QValueKindTimeTZ, QValueKindDate, QValueKindTimestamp, QValueKindTimestampTZ:
-		return &QValueKindAvroSchema{
-			AvroLogicalSchema: map[string]string{
-				"type": "string",
-			},
-		}, nil
+		return "string", nil
 	case QValueKindHStore, QValueKindJSON, QValueKindStruct:
-		return &QValueKindAvroSchema{
-			AvroLogicalSchema: map[string]interface{}{
-				"type":   "string",
-				"values": "string",
-			},
+		return AvroSchemaMap{
```

Comment:
This is the only change in behavior, should be split out to a separate PR or reverted

Reply:
https://github.com/PeerDB-io/peerdb/issues/847